src/components/CardItem/CardItem.tsx

```diff
@@ -0,0 +1,172 @@
+import styled from "styled-components";
+import "react-loading-skeleton/dist/skeleton.css";
+import Delorean from "../../shared/assets/images/delorean.jpg";
+import { BaseColor } from "../../config/color";
+import { Cars, User_Cars } from "../../shared/graphql/__generate__/generated";
+import { Link } from "react-router-dom";
+import { addOneDay } from "../../shared/types/Date";
+import FavoriteButton from "../../pages/ViewCarDetail/FavoriteButton";
+import { useState } from "react";
+
+interface Props {
+  data: Cars;
+  favorite_cars: User_Cars[];
+  showFavorites?: boolean;
+  loadingFavoriteCar?: boolean;
+}
+
+export const CardItem = ({
+  data,
+  favorite_cars,
+  showFavorites = false,
+  loadingFavoriteCar = false,
+}: Props) => {
+  const favoriteCar = favorite_cars.find((item) => item.car_id === data.id);
+
+  const [isFavoriteCar, setIsFavoriteCar] = useState<boolean>(
+    favoriteCar ? true : false
+  );
+
+  if (showFavorites && !isFavoriteCar) {
```

Comment:
I assume you added this cause you are iterating a list, if so I'd suggest to add this condition where the iterator is